### PR TITLE
Add docify github action to help ensure we are patching docstrings correctly

### PR DESCRIPTION
**Pull request by gramster**
Also added some more sanity checking in docify itself so it will be more inclined to fail if we patched a docstring in a bad location.

### CODE REVIEW

File: .github/workflows/docify.yml [hunk 1]
```diff
@@ -0,0 +1,42 @@
+name: 'Docify'
+
+on: [push, pull_request]
```
Comment (bschnurr) [inline .github/workflows/docify.yml:3]: there is a file filter if you want to restrict it to subfolder changes

Comment (bschnurr) [reply .github/workflows/docify.yml:3]: https://docs.github.com/en/actions/using-workflows/workflow-syntax-for-github-actions#example-including-paths

Comment (bschnurr) [reply .github/workflows/docify.yml:3]: oh nevermind. this is python-type-stubs

Comment (gramster) [reply .github/workflows/docify.yml:3]: Well, that's still potentially useful - we should only need to run this if the stubs change, docify changes, the config file changes, or the version of pandas changes. But given we're mostly changing pandas its probably fine to leave as is for now :-)

**Review by bschnurr — APPROVED** [review]

**Pull request merged**